src/custom/hooks/useFetchProfile.ts

```diff
@@ -3,22 +3,51 @@ import { useEffect, useState } from 'react'
 import { getProfileData } from 'api/gnosisProtocol'
 import { ProfileData } from 'api/gnosisProtocol/api'
 
-export default function useFetchProfile() {
+type FetchProfileState = {
+  profileData: ProfileData | null
+  error: string
+  isLoading: boolean
+}
+
+const emptyState: FetchProfileState = {
+  profileData: null,
+  error: '',
+  isLoading: false,
+}
+
+const FETCH_INTERVAL_IN_MINUTES = 5
+
+export default function useFetchProfile(): FetchProfileState {
   const { account, chainId } = useActiveWeb3React()
-  const [profileData, setProfileData] = useState<ProfileData | null>(null)
+  const [profile, setProfile] = useState<FetchProfileState>(emptyState)
 
   useEffect(() => {
+    setProfile({ ...emptyState, isLoading: true })
     async function fetchAndSetProfileData() {
-      if (chainId && account) {
+      if (!chainId || !account) {
+        setProfile((prevState: FetchProfileState) => {
```

Comment:
Also, since the function is pretty simple you could refactor it to not one line without the `return` statement:

```suggestion
        setProfile((prevState) => ({...prevState, isLoading:false}))
```